### DISCUSSION TITLE
Freeing memory in Example1

### DISCUSSION THREAD

**Discussion by goodintention** [FactoryXCode/MfPack#71] [Q&A] [answered]
Thank you for providing this wonderful package,
On Example1, when I choose MP4 output format, the app’s memory usage increases by +59 MB every time I click the ‘Execute’ button, 
How can I free the memory after encoding? 
I’m using the 32-bit Delphi 10.1 Berlin compiler.
Thanks.

**Comment by FactoryXCode** [comment, marked as answer]
Hi,

I made some changes in the sample.

The main issue was that FSampleSinkWriter was not correctly freed from memory.
The others are creating and freeing the sinkwriter in combination with repeatetly calling MFStartup increases memory consumption while running a lot.  This will not affects memory when closing the app, but should not happen.

The changes in the sample are uploaded to GitHub.

Thank you for reporting this issue.

**Comment by goodintention** [comment]
Thank you for the changes, 
now there’s no memory issue, but the example sometimes freezes.

**Reply by FactoryXCode** [reply]
Hi,
On what codeline  does the app freeze exactly?

**Reply by goodintention** [reply]
Hi, 
when I comment out this line: 
//hr := pRintWriter.Flush(stream); 
the app stops freezing.

**Reply by FactoryXCode** [reply]
I remember there are some issues using the Flush method.
See discussion item [Memory usage with ReadSample](https://github.com/FactoryXCode/MfPack/discussions/14)
Thank you and regards, Tony.
